### PR TITLE
Fixed empty machineID issue and enhanced print statements

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:
Fixes issues on machine-controller-manager while integrating with `cmi-client` branch.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: pkg/controller/machine.go [hunk 1]
```diff
@@ -492,20 +492,17 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.CMID
...
-				if err == nil || status.Code(err) == codes.Unimplemented {
-					// Either there is no err
-					// or the ShutDownMachine is not implemented
-					// then continue
-					err = driver.DeleteMachine(machineID)
-				}
+			_, err = driver.GetMachine(machineID)
+			if err != nil && status.Code(err) != codes.Unimplemented {
+				// If GetMachine was implemented and error occurred
+				glog.V(2).Infof("Get Machine called failed with the following error: %s", err)
```
Comment (hardikdr) [inline pkg/controller/machine.go]: `GetMachine call failed` -  typo ?

File: pkg/controller/machine.go [hunk 2]
```diff
@@ -492,20 +492,17 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.CMID
...
+			}
+			err = driver.ShutDownMachine(machineID)
+			if err != nil && status.Code(err) != codes.Unimplemented {
+				// If ShutDownMachine was implemented and error occurred
+				glog.V(2).Infof("Shutdown Machine called failed with the following error: %s", err)
```
Comment (hardikdr) [inline pkg/controller/machine.go]: typo, same as above.

File: pkg/controller/machine.go (hunk 1 above)
Comment (prashanth26) [reply pkg/controller/machine.go]: Fixed.

File: pkg/controller/machine.go (hunk 2 above)
Comment (prashanth26) [reply pkg/controller/machine.go]: Fixed

**Pull request merged**